plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -538,8 +538,7 @@ applyHint ide nfp mhint =
         liftIO $ withSystemTempFile (takeFileName fp) $ \temp h -> do
             hClose h
             writeFileUTF8NoNewLineTranslation temp oldContent
-            (pflags, _, _) <- runAction' $ useNoFile_ GetHlintSettings
```

Comment:
not sure where that falgs can come on but, coud we set by the user in the lsp config, should not we honour them in that case, independently of the actual extension (and even if that make the thing go wrong)?

if setting comes (also) from hlint default ones, would not the plugin and standalone hlint cli diverge in their behaviour (even if the plugin one is better)?

Reply:
> would not the plugin and standalone hlint cli diverge in their behaviour (even if the plugin one is better)?

That's an interesting point. I think it would mean that the plugin version was superior. Maybe that's okay? :thinking: Perhaps so long as we document it?

Reply:
maybe @ndmitchell could help us here

Reply:
Now that I look at this again there is a weak way of figuring out whether we're using `hlint`'s default extensions, or the user has passed it in, and that is by checking whether the `parsedFlags` enabled extensions matches the default extensions list, and whether the disabled extensions list is empty.

I feel like the ultimate goal of `hlint` is to provide accurate suggestions and if they could know the extensions that each file has enabled they wouldn't have provided the option to turn on or off suggestions in the first place.

I think the only reason why a user would have extensions in their hlint config contradicting one of their files is to make another file work, or to test hlint itself.

So... my vote is to just use the extensions provided by mod summary, and document it in the hlint plugin readme.

Reply:
Having said that I think that the hlint plugin already ignores the extensions from the hlint config when HLINT_ON_GHC_LIB is not defined, so we're just matching the functionality. This seems to affect ghc 8.8.4 and lower.

Reply:
Fair enough, last argument is definitive